### DISCUSSION TITLE
Fresh Install of Master No /root/master_setup dir created post install

### DISCUSSION THREAD

**Discussion by jdpadro** [Security-Onion-Solutions/securityonion#4337] [Unsupported Versions] [answered]
So I finally decided to move my distributed IDS architecture from 16.04 to SO2. Since a master must be present for the sensors to connect to during their setup process. I decided to work on the master. I downloaded the 2.3.50 ISO image. Since that image is +7G I had to create a bootable USB. Popped it onto the server rebooted the device selected the usb device as my boot device and away it goes. Created the admin account and password finished the first stage install. Rebooted the device logged in using admin account created and proceeded to second stage installation, selected Distributed, master+searchnode, configured the network card and completed this stage successfully. Rebooted the server once again and performed the "soup" which creates the required certificates and updates the install to latest version, completed this stage. Rebooted the server and then connected the analyst workstation to the server via web interface of master. Next decided to work on the first sensor, performed the first and started the second stage, during this stage an attempt is made to connect to the master, the certificate was accepted and a successful connection was accomplished. After a successful login the next step seems to be to launch the so-setup scrippt for the sensor from the master. Instead what was returned was:
 bash: /root/master_setup/securityonion/setup/so-setup: No such file or directory

At this point the installtion exits and I am returned to the sensors command prompt. 

I looked at the master for this directory and it was not there. Any thoughts on what happened and how to correct this issue?

So I have reinstalled using the same 2.3.50 ISO and the problem of the missing /root/master_setup/ directory still persists. 

Thanks

**Comment by rwwiv** [comment, marked as answer]
A little background: that directory (`/root/manager_setup`) is generated on the sensor (not the manager) during setup by cloning a tarball from the manager that _should_ have contained the contents of the most recent version of the git repo. This is to prevent version mismatch when running a different version of setup on a forward node than what is on the manager.

However, there was a bug in the soup script that meant the updated tarball generated from the cloned repo wasn't generated correctly. This will be fixed in an upcoming release, but in the meantime you can copy the below bash script to your manager and run it as root to fix the problem.


------
Run using `sudo bash <script_filename>`

```sh
#!/bin/bash

main() {
  set -e
  local tar_dir="/tmp/repo_tar_fix"
  mkdir -p "$tar_dir"
  cd "$tar_dir" && rm -rf ./*

  git clone --quiet https://github.com/Security-Onion-Solutions/securityonion.git 1> /dev/null
  cd securityonion

  local cloned_version
  cloned_version=$(cat ./VERSION)
  local current_version
  current_version=$(cat /etc/soversion)

  echo "Checking if cloned version matches current version..."
  if [[ $cloned_version != "$current_version" ]]; then
    echo "Version mismatch, please run soup first."
    exit 1
  else
    echo "Versions match, proceeding."
  fi

  tar -czf "/opt/so/repo/$cloned_version.tar.gz" -C "$tar_dir/securityonion" .

  set +e
}

echo "Fixing repo tarball on manager..."
main
echo "Fix applied, now safe to run setup on forward nodes."

```

**Reply by jdpadro** [reply]
Thank you for the response I will give this a whirl. Now how can I restart the install on the sensor that attempted the connection to the master so it will make that attempt again. Currnelty if I login to that sensor it will continue the install from itself and not intiate the connection to the master.

**Reply by rwwiv** [reply]
I should have made it clear in my first response, you'll need to delete the `/root/manager_setup` directory on the sensor before restarting setup. 

As far as how to actually restart setup, if you're unable to cancel out of of the menus the easiest way to restart setup would be to just restart the sensor itself. Once you do so, you'll want to log back in as the admin user you set up. 

After you've either cancelled setup or logged back in after a reboot, just run the following commands as the admin user:
```sh
sudo rm -rf /root/manager_setup # remove the old downloaded setup files
```
```sh
cd ~/SecurityOnion/setup; sudo bash so-setup iso # re-run the iso setup
```

**Reply by jdpadro** [reply]
Thanks William for that I will proceed as stated.

As for the fix_master script that you provided. I did copy the contents from the earlier post and proformed the following;
As the root user 
     cd /root
     vi fix_master_setup.sh
        Pasted contents into this file and :wq to exit and save. 
      chmod +x fix_master_setup.sh
      tried to execute with 
          bash fix_master_setup.sh  nothing happens no output no directory created. 
          sh fix_master_setup.sh  noting happens no directory created
          ./fix_master_setup.sh  nothing happens no directory created

  When I perform the execute with the -v it outputs the contents of the script verbatim.

Any suggestions

Thanks

**Reply by rwwiv** [reply]
Sorry, I'm not able to replicate that behavior on my end. On a test server just now I was able to (as root) run the following commands and the script executed fine:
```sh
cd /root
vi fix_master_setup.sh
  # Type i to enter INSERT mode
  # Paste contents of script
  # Quit and write the file using :wq
chmod +x fix_master_setup.sh
./fix_master_setup.sh
```
